### PR TITLE
boards: add support for Adafruit Feather nRF52840 Sense

### PR DESCRIPTION
**Pull request by MichelRottleuthner**
### Contribution description

This adds support for the [Adafruit Feather nRF52840 Sense](https://learn.adafruit.com/adafruit-feather-sense). A nice prototyping board that comes with many onboard sensors. RIOT already supports most of it.

Note: you may need to [update the bootloader](https://learn.adafruit.com/adafruit-feather-sense/update-bootloader) in order to make flashing work. Version 0.7 worked fine for me. Note that some OSes do not auto mount the UF2 drive which prevents flashing. The uf2 helper `dist/tools/uf2/uf2conv.py` could use some improvements for handling distro-specific mount paths properly, which also applies to existing boards.

#### Unsupported features
Making use of the LSM6DS33 gyro/accelerometer might need some work on an existing driver (https://github.com/RIOT-OS/RIOT/issues/17053). Currently there is no support for the PDM peripheral used by the [digital MEMS microphone](https://www.st.com/resource/en/datasheet/mp34dt01-m.pdf). The NeoPixel LED lacks an nrf52-specific integration for the `ws281x` driver. All of that can be done in follow ups.

### Testing procedure
Flash and run some apps e.g., `examples/gnrc_networking` and `examples/saul` and check that both work as intended.
In the saul app `> saul read all` should give you a bunch of correct readings for the sensors. Both worked fine for me.

### Issues/PRs references
None.

### CODE REVIEW

File: boards/feather-nrf52840-sense/doc.txt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+![top-down view on feather-nrf52840-sense][top-down view]
+
+[feather-nrf52840]: https://learn.adafruit.com/introducing-the-adafruit-nrf52840-feather/
+[feather-nrf52840-sense]: https://learn.adafruit.com/adafruit-feather-sense
+[top-down view]: https://cdn-learn.adafruit.com/assets/assets/000/089/096/original/sensors_Feather_Sense_top.jpg
+
+### Flashing, Bootloader, and Terminal
+
+Refer to [The Feather nRF52840 Express
+documentation](https://doc.riot-os.org/group__boards__feather-nrf52840.html) for furhter details.
```
Comment (benpicco) [inline boards/feather-nrf52840-sense/doc.txt:24]:
```suggestion
documentation](https://doc.riot-os.org/group__boards__feather-nrf52840.html) for further details.
```

**Review by benpicco — APPROVED** [review]
Looks good!

File: boards/feather-nrf52840-sense/Makefile.dep [hunk 2]
```diff
@@ -0,0 +1,11 @@
+ifneq (,$(filter saul_default,$(USEMODULE)))
+  USEMODULE += apds9960
+  USEMODULE += bmp280_i2c
+  USEMODULE += lis3mdl
```
Comment (benpicco) [inline boards/feather-nrf52840-sense/Makefile.dep:4]: didn't you write the LSM6DS33 does not work with this driver?

File: boards/feather-nrf52840-sense/doc.txt (hunk 1 above)
Comment (MichelRottleuthner) [reply boards/feather-nrf52840-sense/doc.txt]: Done.

File: boards/feather-nrf52840-sense/Makefile.dep (hunk 2 above)
Comment (benpicco) [reply boards/feather-nrf52840-sense/Makefile.dep:4]: Ah nvm lis3mdl is a magnetometer

Comment (MichelRottleuthner) [reply boards/feather-nrf52840-sense/Makefile.dep:4]: Yeah but LSM6DS33 != LIS3MDL ;) The existing `lsm6dsl` driver can probably be adjusted for the LSM6DS33 (as per the issue I linked above). But I guess that is out of scope for this PR.

**Comment by MichelRottleuthner** [comment]
Amended missing kconfig sensor declarations to fix make/kconfig module mismatch.

**Comment by benpicco** [comment]
bors merge

**Comment by MichelRottleuthner** [comment]
Thanks for the swift review @benpicco

**Pull request merged**